apollo-ios/Sources/Apollo/JSONConverter.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+#if !COCOAPODS
+import ApolloAPI
+#endif
+
+public enum JSONConverter {
+  
+  /// Converts a ``SelectionSet`` into a basic JSON dictionary for use.
+  ///
+  /// - Returns: A `[String: Any]` JSON dictionary representing the ``SelectionSet``.
+  public static func convert(_ selectionSet: some SelectionSet) -> [String: Any] {
```

Comment:
as you might have guessed from my PR, I'm not a fan of relying on the compiler to select the right overload, but that's my personal preference. The changes in this PR still work fine for my use case. Thanks for the help!